### ISSUE TITLE
react/display-name false positive in render prop ternary

### ISSUE THREAD

**Issue by jaydenseric** [yannickcr/eslint-plugin-react#2224]
The `react/display-name` rule has a false positive error when a component render prop is a ternary containing an arrow function expression:

```jsx
<Section renderFigure={image ? () => <img src={image.src} /> : null} />
```

This does not have an error:

```jsx
<Section renderFigure={() => <img src={image.src} />} />
```

So the ternary must confuse the rule somehow.

<img width="558" alt="Screen Shot 2019-04-03 at 4 02 16 pm" src="https://user-images.githubusercontent.com/1754873/55454012-eceb4400-5629-11e9-8a36-f767a1dfa84a.png">

**Comment by ljharb** [comment]
In this case, the arrow function is considered a component.

It seems worth improving our component detection to avoid render props - cc @alexzherdev

**Comment by pawelnvk** [comment]
Just a side-note, but I would rather expect ternary inside of callback instead of passing callback if ternary is fulfilled.

**Comment by jzabala** [comment]
This doesn't happen with versions:

```
"babel-eslint": "^10.1.0"
"eslint": "^7.4.0"
"eslint-plugin-react": "^7.20.3"
```

![Screen Shot 2020-07-10 at 6 33 04 PM](https://user-images.githubusercontent.com/1315054/87207978-518c5880-c2db-11ea-968c-db9aa6d064e4.png)

**Comment by ljharb** [comment]
@jzabala your screenshot shows that the false positive does happen?

**Comment by jzabala** [comment]
> @jzabala your screenshot shows that the false positive does happen?

Sorry for the confusing screenshot. I wanted to show that the rule was active, since it is triggered at the bottom, but it is not triggered in the case reported in the issue (the ternary).

**Comment by ljharb** [comment]
ah, thanks. Closing, then :-)
